usr/share/rear/backup/RSYNC/GNU/Linux/620_force_autorelabel.sh

```diff
@@ -1,3 +1,5 @@
+local backup_prog_rc
+
 [ -f $TMP_DIR/force.autorelabel ] && {
 
 	> "${TMP_DIR}/selinux.autorelabel"
```

Comment:
@pcahyna 
a question out of curiosity:
Could you explain what code like `> .../selinux.autorelabel` is meant to do?
I am not a SELinux user (at least I don't use it knowingly)
so for me that code look rather obscure.
My guess is that creating an empty `selinux.autorelabel` file
in whatever "right" directory triggers something - but what?

Reply:
It triggers selinux relabeling after boot of the restored system. The primary source of the SELinux information on files is some database that assigns contexts to file paths. But this information is not directly used by the kernel, the kernel checks the `security.selinux` extended attribute on the given file. So if the extended attributes get out of sync with the database (this can happen because of restoring a backup that does not contain this information), some tool needs to set the extended attributes again on all the files that should have them, and this is called relabeling.

Now, in ReaR, it is hard to understand how it is done, because there are several files to trigger it, with a similar functionality, like `force.autorelabel` and `selinux.autorelabel` and it is not obvious how they interact. I also don't see why it has to be backup method specific, because it means that every backup method needs to reimplement it (most likely most of them need it). I think having it in output instead of backup, like the bootloader information and disk layout, would be cleaner. I have not investigated much because I am afraid it will be another can of worms.

Reply:
The `selinux.autorelabel` is an old flag file to trigger relabeling the files with SELinux after the restore happened. It was meant only for internal backup mechanism like tar and rsync. External backup flows should have implemented the restore of the SELinux labels by their software (we just don't care about these flows as we are not responsible of the backups made by external backup software).
By default (for internal backup flows) we turn off selinux during the backup procedure. However, we hope that 10 years later tar and rsync would have been enough involved to make backups with the keeping the label structure, no?

Reply:
> External backup flows should have implemented the restore of the SELinux labels by their software

I am not at all sure that external backup software is able to save and restore this kind of metadata. That's why I have thought that it might be better to implement relabeling generally in output. But maybe I am too pessimistic.

> By default (for internal backup flows) we turn off selinux during the backup procedure. 

I am curious why is it needed to turn off selinux during the backup procedure. Even if the backup tool does not have the support for storing selinux metadata, I would not expect that having selinux on makes the problem worse, or does it?

> However, we hope that 10 years later tar and rsync would have been enough involved to make backups with the keeping the label structure, no?

I suppose so, but I will need to test this.